src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala

```diff
@@ -618,4 +617,31 @@ final case class EServicesApiServiceImpl(
       }
     }
   }
+
+  override def deleteEService(
+    eServiceId: String
+  )(implicit contexts: Seq[(String, String)], toEntityMarshallerProblem: ToEntityMarshaller[Problem]): Route = {
+
+    def deleteEserviceIfEmpty(eService: CatalogProcessEService, isDraftDeleted: Boolean = true): Future[Unit] =
+      if (eService.descriptors.isEmpty) {
+        catalogProcessService.deleteEService(eService.id)
+        Future.successful(()) // Adding this in case the eService is already empty in order to not repeat the delete
```

Comment:
careful here, the result of the first `Future` is discarded.
Can you give more details about this comment?
I would expect just
```suggestion
        catalogProcessService.deleteEService(eService.id)
```

Reply:
You're absolutely right, it's not needed because the deleteEService already returns a success or a fail.